blog/2024-02-26-Get-Started-With-Uptane/index.md

```diff
@@ -0,0 +1,112 @@
+
+So, once you have a director repository, an image repository, and a device, the picture is complete. In broad strokes, the protocol for checking for an update goes like this:
+
+![](uptane-repository.png)
+
+
+That's more or less it. Of course, there are many, many details. The bulk of the standard relates to the exact procedures for how the device checks the metadata and signatures. And of course, we've ignored the question of how software gets into a repository, how the signing keys are managed, and how Director decides what should be installed on the device. Uptane also deals with "secondaries", meaning that one internet-connected microcontroller or embedded linux device can download and validate firmware updates for other microcontrollers; you can read elsewhere about how secondaries work.
+
+### What Uptane Isn't
+
+If you've gotten to this point, you've probably realized that there is a significant amount of stuff missing from the picture. Uptane doesn't say anything about:
+
+* an interface to manage the software in the repository
+* how to keep track of the information Director gets from the devices
+* how Director decides what software the device should have installed
+* how the device actually installs the update
+* how the device authenticates itself to the repository
+* transport security between the repository and the device (though Uptane is resilient to transport security compromise)
+
+So there's a lot of other stuff that you need to attach Uptane concepts to before you have a complete update system. Some of those things might be relatively easy, some harder, but there are certainly a lot of them.
+
+The reason Uptane was created this way is that it was written to solve real problems with existing systems. TUF was created as a way for existing software repositories to improve their security; those repositories already had systems for developers to register their package names and upload their software, and clients that know how to install the packages. Uptane was created to be used by automakers who already have software repositories, suppliers sending them firmware, quality control processes for what constitutes a software release, aftersales and maintenance departments keeping track of what parts might need software updates, and so on. It's a standard that answers the question, "How can I make my existing, complex system safer?" It would also make the standard very bloated and very inflexible if we made arbitrary decisions about those auxiliary pieces; we want the Uptane standard to remain focused on solving supply-chain security and software deployment security.
```

Comment:
Capitalization of the Standard. If mentioned as a proper title, as it is later in this paragraph, Standard should be capitalized.

```suggestion
The reason Uptane was created this way is that it was written to solve real problems with existing systems. TUF was created as a way for existing software repositories to improve their security; those repositories already had systems for developers to register their package names and upload their software, and clients that know how to install the packages. Uptane was created to be used by automakers who already have software repositories, suppliers sending them firmware, quality control processes for what constitutes a software release, aftersales and maintenance departments keeping track of what parts might need software updates, and so on. It's a standard that answers the question, "How can I make my existing, complex system safer?" It would also make the standard very bloated and very inflexible if we made arbitrary decisions about those auxiliary pieces; we want the Uptane Standard to remain focused on solving supply-chain security and software deployment security.
```